okio/src/commonMain/kotlin/okio/Path.kt

```diff
@@ -249,6 +249,12 @@ class Path private constructor(
     return buffer.toPath(directorySeparator = slash)
   }
 
+  override fun compareTo(other: Path): Int {
```

Comment:
This and equals/hashCode probably requires a comment somewhere on case sensitivity.  Do we have that already.

Reply:
Lemme add it.

Our rule is very straightforward: only the OS can do this correctly.

On my Mac I have a case-sensitive volume mounted for whenever I need to do awkward git stuff like filter-branch. The paths `/Volumes/case-sensitive/foo.txt` and `/VOLUMES/case-sensitive/foo.txt` point to the same file, but `/Volumes/case-sensitive/FOO.txt` is a different file.